docs/source/frameworks/onnx.rst

```diff
@@ -0,0 +1,329 @@
+
+   .. code-block:: python
+
+      batch_size = 1 # can be any number
+      # Tracing input to the model
+      x = torch.randn(batch_size, 1, 224, 224, requires_grad=True)
+
+      # Export the model
+      torch.onnx.export(torch_model,
+			x,
+			"super_resolution.onnx",   # where to save the model (can be a file or file-like object)
+			export_params=True,        # store the trained parameter weights inside the model file
+			opset_version=10,          # the ONNX version to export the model to
+			do_constant_folding=True,  # whether to execute constant folding for optimization
+			input_names = ['input'],   # the model's input names
+			output_names = ['output'], # the model's output names
+			dynamic_axes={'input' : {0 : 'batch_size'},    # variable length axes
+				      'output' : {0 : 'batch_size'}})
+
+   Now we can compute the output using ONNX Runtime’s Python APIs:
+
+   .. code-block:: python
+
+      import onnxruntime
+
+      ort_session = onnxruntime.InferenceSession("super_resolution.onnx")
+      # compute ONNX Runtime output prediction
+      ort_inputs = {ort_session.get_inputs()[0].name: to_numpy(x)}
+      # ONNX Runtime will return a list of outputs
+      ort_outs = ort_session.run(None, ort_inputs)
+      print(ort_outs[0])
+
+.. tab:: TensorFlow
+
+   TODO
+
+.. tab:: Scikit Learn
+
+   TODO
+
+
+Saving ONNX model with BentoML
+-----------------------------
+
+To quickly save an ONNX model to BentoML's :ref:`Model
+Store<concepts/model:Managing Models>`, first use ``onnx.load`` to
+load the exported ONNX model back into ``onnx.ModelProto`` object,
+then call BentoML's ``save_model``:
+
+
+.. code-block:: python
+
+   signatures = {
+       "run": {"batchable": True},
+   }
+   bentoml.onnx.save_model("super_resolution", onnx_model, signatures=signatures)
+
+which will result:
+
+.. code-block:: bash
+
+   Model(tag="super_resolution:lwqr7ah5ocv3rea3", path="~/bentoml/models/super_resolution/lwqr7ah5ocv3rea3/")
+
+.. note::
+
+   ``save_model`` will use ``{"run": {"batchable": False}}`` as
+   default signatures if ``signatures`` is not provided. Set
+   ``batchable`` to ``False`` will disable BentoML's
+   :ref:`guides/batching:Adaptive Batching` functionality. That's why
+   we provide our own signatures here. Read more about :ref:`Model
+   Signatures <concepts/model:Model Signatures>` and :ref:`Batch Input
+   <concepts/model:Batching>`
+
+.. seealso::
+
+   ``save_model`` also has some :ref:`general options
+   <concepts/model:Save A Trained Model>` for functionalities like
+   saving metadata and custom objects.
+
+
+Loading an ONNX model with BentoML
+----------------------------------
+
+We can use ``load_model`` to load an ONNX model back to memory:
+
+.. code-block:: python
+
+   ort_session = bentoml.onnx.load_model("onnx_super_resolution")
+
+.. note::
+
+   BentoML will load an ONNX model back as an
+   ``onnxruntime.InferenceSession`` object which is ready to do
+   inference
```

Comment:
Maybe it is nice to have one, two sentences separating the difference between `onnx` and `onnxruntime`.

Reply:
I think we can do that in the preface